Core.Collectors/Web/RateLimiter.cs

```diff
@@ -152,7 +198,14 @@ public async Task UpdateStatsAsync(string identity, string requestUrl, HttpRespo
                 return;
             }
 
-            await this.rateLimiterCache.CacheAsync(new RateLimitTableEntity(identity, this.OrganizationId, this.OrganizationName, rateLimitLimit, rateLimitRemaining, rateLimitResetDate, retryAfterDate)).ConfigureAwait(false);
+            RateLimitTableEntity rateLimitTableEntity = new RateLimitTableEntity(identity, this.OrganizationId, this.OrganizationName, rateLimitLimit, rateLimitRemaining, rateLimitResetDate, retryAfterDate);
+            await this.rateLimiterCache.CacheAsync(rateLimitTableEntity).ConfigureAwait(false);
+            // Dual writing for ADO Collectors is enabled.
+            // This is necessary here as the derived rate limiter class is missing telemetry.
+            if (this.enableDualWriting)
```

Comment:
C# question: If enableDualWriting isn't initialized (i.e. It's called from GitHub collectors) does throw an exception? Or does it return a null value and evaluate to false?

Reply:
I don't understand this question, sorry. How does GitHub collectors call this to disable the dual writing? The additional parameter in the constructor is not optional (e.g., does not have a default), so are you planning to change the GH collectors and ADO collectors accordingly? Regardless, would not this be a backwards compatibility change as things are right now (for sub modules)? 

Do you want to instead, introduce this as an optional parameter with default = false and then in ADO collectors send enableDualWriting: true?

Reply:
Actually I took another look and see that you have introduced complete new constructors, which would work. You can do this way if you want; that's fine.

To answer your C# questions, then yes, this would be no-op for GitHub since it is calling the existing constructor and the default value of a boolean is false. However, you should never rely on default values, instead in the old constructors initialize this.enableDualWriting = false; to ensure the consistent behavior.

Reply:
Done